File: v1/src/main/java/com/api/v1/medical_slots/CancelledDefaultMedicalSlotResponseDTO.java
package com.api.v1.medical_slots;

import com.api.v1.doctors.DoctorResponseDTO;
import com.api.v1.medical_slots.enums.MedicalSlotStatus;

import java.time.LocalDateTime;

public final class CancelledDefaultMedicalSlotResponseDTO extends DefaultMedicalSlotResponseDTO {

    private final LocalDateTime cancelledAt;

    CancelledDefaultMedicalSlotResponseDTO(String id,
                                           MedicalSlotStatus status,
                                           DoctorResponseDTO doctor,
                                           LocalDateTime availableAt,
                                           LocalDateTime createdAt,
                                           LocalDateTime cancelledAt
    ) {
        super(id, status, doctor, availableAt, createdAt);
        this.cancelledAt = cancelledAt;
    }

    public LocalDateTime getCancelledAt() {
        return cancelledAt;
    }
}

File: v2/src/main/kotlin/com/api/v2/medical_slots/CancelledDefaultMedicalSlotResponseDTO.kt
package com.api.v2.medical_slots

import com.api.v2.doctors.DoctorResponseDTO
import com.api.v2.medical_slots.enums.MedicalSlotStatus
import java.time.LocalDateTime

class CancelledDefaultMedicalSlotResponseDTO(
    id: String,
    status: MedicalSlotStatus,
    doctor: DoctorResponseDTO,
    availableAt: LocalDateTime,
    createdAt: LocalDateTime,
    cancelledAt: LocalDateTime
) : DefaultMedicalSlotResponseDTO(id, status, doctor, availableAt, createdAt)